tests/behat/features/web/project-details/project_description.feature

```diff
@@ -41,6 +41,150 @@ Feature: Projects should have descriptions that can be changed by the project ow
+    Given I log in as "Catrobat"
+    And I go to "/app/project/1"
+    And I wait for the page to be loaded
+    Then the element "#edit-description-button" should be visible
+    When I click "#edit-description-button"
+    And I wait for AJAX to finish
+    Then the element "#description" should not be visible
+    But the element "#edit-description" should be visible
+    And the element "#description-language-selector" should be visible
+    And the element "#edit-description-submit-button" should be visible
+    Then I choose "French" from selector "#description-language-selector"
+    And I wait for AJAX to finish
+    Then I fill in "edit-description" with "This is a description translation"
+    Then I choose "Russian" from selector "#description-language-selector"
+    And I should see "Would you like to keep your current changes?"
+    When I click ".swal2-close"
+    And I should see "This is a description translation"
+    And I should see "French"
+  
+  Scenario: Saving changes after closing description editor
+    Given I log in as "Catrobat"
+    And I go to "/app/project/1"
+    And I wait for the page to be loaded
+    Then the element "#edit-description-button" should be visible
+    When I click "#edit-description-button"
+    And I wait for AJAX to finish
+    Then the element "#description" should not be visible
+    But the element "#edit-description" should be visible
+    And the element "#description-language-selector" should be visible
+    And the element "#edit-description-submit-button" should be visible
+    Then I fill in "edit-description" with "This is a new description"
+    And I click "#edit-description-close-button"
+    And I should see "Do you want to save your changes?"
+    When I click ".swal2-confirm"
+    Then the element "#description" should be visible
+    And the element "#edit-description-ui" should not be visible
+    And I should see "This is a new description"
+
+  Scenario: Discarding unsaved changes after closing description editor
+    Given I log in as "Catrobat"
+    And I go to "/app/project/1"
+    And I wait for the page to be loaded
+    Then the element "#edit-description-button" should be visible
+    When I click "#edit-description-button"
+    And I wait for AJAX to finish
+    Then the element "#description" should not be visible
+    But the element "#edit-description" should be visible
+    And the element "#description-language-selector" should be visible
+    And the element "#edit-description-submit-button" should be visible
+    Then I fill in "edit-description" with "This is a new description"
+    And I click "#edit-description-close-button"
+    And I should see "Do you want to save your changes?"
+    When I click ".swal2-deny"
+    Then the element "#description" should be visible
+    And the element "#edit-description-ui" should not be visible
+    And I should see "my description"
+
+  Scenario: Going back to unsaved changes after closing description editor
```

Comment:
Editing custom translation, closing the editor but going back to unsaved changes

Reply:
Updated name, but changed it to "Editing description, ..." since this test doesn't edit a custom translation